### DISCUSSION TITLE
Theme Provider or Theme Service for v3

### DISCUSSION THREAD

**Discussion by AathifMahir** [microsoft/fluentui-blazor#639] [General]
Are there plans to develop Theme Provider or Theme Service for v3 or maybe v3.1, I guess it's going to be critical part of Fluent Theming System Going Forward, Maybe i'm mistaken and the feature is already there?

I'm not sure whether Design Tokens were the solution for multiple different layouts in the Blazor.

**Comment by vnbaaij** [comment]
No, there are no plans for that. Don't think it is a critical part as we 'only' follow the Fluent Design. Theming is already done in the components themselves. It is possible to tweak things to specific needs by means of the Design Token implementation. Examples of that can be found in the demo sites.

But interested to hear what others think though. Moving this to the discussion area as it more in its place there.

**Comment by AathifMahir** [comment]
> No, there are no plans for that. Don't think it is a critical part as we 'only' follow the Fluent Design. Theming is already done in the components themselves. It is possible to tweak things to specific needs by means of the Design Token implementation. Examples of that can be found in the demo sites.
> 
> But interested to hear what others think though. Moving this to the discussion area as it more in its place there. 

In my case, I'm really facing difficulty in making the Fluent theming to work on multiple layouts in Blazor. Currently I have 3 layouts, which are abstract BaseLayout which Initializes Theme And Main layout and Empty Layout, Doing so most cases I can't seem to reliably apply the theme for the empty layout.

**Reply by vnbaaij** [reply]
Can you elaborate on what is in your 'theme' then? I think your scenario is very well possible to achieve with a couple of Design Tokens

**Reply by AathifMahir** [reply]
> Can you elaborate on what is in your 'theme' then? I think your scenario is very well possible to achieve with a couple of Design Tokens 

Let me explain what exactly Theme Provider Does.

Let's say for example:
We have IThemeProvider that Injected to DI

Which has a couple of methods and Properties Keeps the Theme Reference by doing the Theming.

Let's say SetTheme(), SetColor() Methods and IsDarkMode, Selected color, Reference

In this Case, Reference is an Element reference which is globally accessible via DI

And others Do the Job at Single Line of Code. That's all

Having a simple IThemeProvider does makes life simpler instead going through lots of hassle on injecting multiple services does the same. As you can this is just an abstraction to make life easier.

Another point on doing this, having to currently selected Theme and Color and etc.. in Local Storage. Therefore we can make that theme persistent for scoped

**Comment by vnbaaij** [comment]
Ok, I understand better now. Yes, think this is something we can work on for a next version.
